src/dmd/ctfe/bc_common.d

```diff
@@ -0,0 +1,1211 @@
+/**
+ * This module is for common utilities for the bytecode engine.
+ *
+ * Copyright:   Copyright (C) 2022 by The D Language Foundation, All Rights Reserved
+ * Authors:     Stefan Koch, Max Haughton
+ */
+module dmd.ctfe.bc_common;
+
+import dmd.ctfe.fpconv_ctfe;
+
+/// functions with index skipFn will be skipped
+/// calling them is equivlent to an expensive nop
+/// this is true for direct and indirect calls
+enum skipFn = uint.max;
+enum currentScope = uint.max - 2;
+
+struct CndJmpBegin
+{
+    BCAddr at;
+    BCValue cond;
+    bool ifTrue;
+}
+
+const(uint) align4(const uint val) @safe pure @nogc
```

Comment:
these first couple of functions feel like they should be part of `root` or `common`, the `static`s should probably be private.